gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/rest/api/service/v4/impl/validation/ApiValidationServiceImplTest.java

```diff
@@ -282,6 +292,43 @@ public void shouldNotChangeLifecycleStateFromCreatedInReview() throws TechnicalE
         );
     }
 
+    @Test
+    public void canDeployWithOnePublishedPlan() {
+        final ExecutionContext executionContext = GraviteeContext.getExecutionContext();
+        final String apiId = "api-id";
+
+        when(planService.findByApi(executionContext, apiId)).thenReturn(Set.of(new PlanEntity().withStatus(PlanStatus.PUBLISHED)));
+        assertTrue(apiValidationService.canDeploy(executionContext, apiId));
+    }
+
+    @Test
+    public void canDeployWithOneDeprecatedPlan() {
+        final ExecutionContext executionContext = GraviteeContext.getExecutionContext();
+        final String apiId = "api-id";
+
+        when(planService.findByApi(executionContext, apiId)).thenReturn(Set.of(new PlanEntity().withStatus(PlanStatus.DEPRECATED)));
+        assertTrue(apiValidationService.canDeploy(executionContext, apiId));
+    }
+
+    @Test
+    public void cannotDeployWithNoPlan() {
+        final ExecutionContext executionContext = GraviteeContext.getExecutionContext();
+        final String apiId = "api-id";
+
+        when(planService.findByApi(executionContext, apiId)).thenReturn(Set.of());
+        assertTrue(apiValidationService.canDeploy(executionContext, apiId));
+    }
+
+    @Test
+    public void canDeployWithNoActivePlan() {
+        final ExecutionContext executionContext = GraviteeContext.getExecutionContext();
+        final String apiId = "api-id";
+
+        when(planService.findByApi(executionContext, apiId))
+            .thenReturn(Set.of(new PlanEntity().withStatus(PlanStatus.STAGING), new PlanEntity().withStatus(PlanStatus.CLOSED)));
+        assertTrue(apiValidationService.canDeploy(executionContext, apiId));
```

Comment:
Shouldn't we test false result here ?
```suggestion
        assertFalse(apiValidationService.canDeploy(executionContext, apiId));
```